iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/ParamTreeImpl.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.docker.tree.impl;
+
+import java.util.List;
+import org.sonar.iac.common.api.tree.Tree;
+import org.sonar.iac.docker.tree.api.ParamTree;
+import org.sonar.iac.docker.tree.api.SyntaxToken;
+
+public class ParamTreeImpl extends DockerTreeImpl implements ParamTree {
+
+  private final SyntaxToken prefix;
+  private final SyntaxToken name;
+  private final SyntaxToken equals;
+  private final SyntaxToken value;
+
```

Comment:
Look very much like the new version of KeyValuePair in PR ADD, it has the almost same parameters prefix / key / equals / value (most of them optional).

I guess we could either switch to it or keep KeyValuePair for cases without prefix and use ParamTree in ADD.

Reply:
I would like to replace all `KeyValuePair` usage with a more precise `ParamTree` where applicable. However, there are cases, where we will need this `KeyValuePair`.